engine/client/cl_scrn.c

```diff
@@ -57,13 +57,22 @@ void SCR_DrawFPS( void )
 	static int	minfps = 9999;
 	static int	maxfps = 0;
 	double		newtime;
-	char		fpsstring[64];
+	char		fpsstring[2048];	//char fpsstring[64]; // Heap allocation if 2048 too much ?
```

Comment:
I recommend you moving this code to separate function, so SCR_DrawFPS stays only for showing FPS information. One function does one thing.

Reply:
It would be good to have OOP used here but since it can't be, copy-paste will be required for most of the SCR_DrawFPS code (to the new function). I would not favor copying and pasting because it makes it spaghetti-code but I could not find another good solution.

Reply:
OOP is possible for C too. Partially.

I planned to share FPS stats between functions, so net_graph can show precise FPS too, instead of silly `1 / host.frametime`.

So we would have a function for FPS calculation(current, best, worst and average), which writes result to public structure, and it can be used by "informative" code, like DrawFPS or Network Graph.